packages/arb-provider-go/test/connection_test.go

```diff
@@ -0,0 +1,280 @@
+		}
+	case err := <-errChan:
+		return err
+	}
+
+	// start RPC server VM
+	server := coordinator.StartRPCServerVM(coord)
+
+	// Run server
+	s := rpc.NewServer()
+
+	s.RegisterCodec(json.NewCodec(), "application/json")
+	s.RegisterCodec(json.NewCodec(), "application/json;charset=UTF-8")
+
+	if err := s.RegisterService(server, "Validator"); err != nil {
+		return err
+	}
+	r := mux.NewRouter()
+	r.Handle("/", s).Methods("GET", "POST", "OPTIONS")
+	//attachProfiler(r)
+
+	headersOk := handlers.AllowedHeaders([]string{"X-Requested-With", "Content-Type", "Authorization"})
+	originsOk := handlers.AllowedOrigins([]string{"*"})
+	methodsOk := handlers.AllowedMethods([]string{"GET", "HEAD", "POST", "PUT", "OPTIONS"})
+
+	go func() {
+		err = http.ListenAndServe(":1235", handlers.CORS(headersOk, originsOk, methodsOk)(r))
+		if err != nil {
+			t.Errorf("ListenAndServe error %v", err)
+		}
+	}()
+
+	return nil
+
+}
+
+func _computePubKeyString(privKeyBytes []byte) (string, error) {
+	privKey, err := crypto.ToECDSA(privKeyBytes)
+	if err != nil {
+		return "", err
+	}
+	pubKey := privKey.Public().(*ecdsa.PublicKey)
+	buf := crypto.FromECDSAPub(pubKey)
+	return hexutil.Encode(buf), nil
+}
+
+func RunValidators(t *testing.T) (*FibonacciSession, error) {
+	coordinatorKey := "ffb2b26161e081f0cdf9db67200ee0ce25499d5ee683180a9781e6cceb791c39"
+	followerKey := "979f020f6f6f71577c09db93ba944c89945f10fade64cfc7eb26137d5816fb76"
+	err := setupValidators(coordinatorKey, followerKey, t)
+	if err != nil {
+		t.Errorf("Validator setup error %v", err)
+		t.FailNow()
+		return nil, err
+	}
+	//setupValidators()
+
+	privateKeyBytes, _ := hex.DecodeString(coordinatorKey)
+	pubKey, err := _computePubKeyString(privateKeyBytes)
+	if err != nil {
+		t.Errorf("_computePubKeyString error %v", err)
+		return nil, err
+	}
+	conn, dialerr := goarbitrum.Dial("", privateKeyBytes, pubKey)
+	if dialerr != nil {
+		t.Errorf("Dial error %v", dialerr)
+		return nil, err
+	}
+	key1, err := crypto.HexToECDSA(coordinatorKey)
+	if err != nil {
+		t.Errorf("HexToECDSA error %v", err)
+		return nil, err
+	}
+	auth := bind.NewKeyedTransactor(key1)
+	auth.GasLimit = 100000000
+
+	//conn, err = goarbitrum.Dial("", privateKeyBytes, _computePubKeyString(privateKeyBytes))
+	var fibAddr common.Address
+	fibAddr = common.HexToAddress("0x895521964D724c8362A36608AAf09A3D7d0A0445")
+	fib, err := NewFibonacci(fibAddr, conn)
+	if err != nil {
+		t.Errorf("NewFibonacci error %v", err)
+		return nil, err
+	}
+
+	//Wrap the Token contract instance into a session
+	fibonacciSession := &FibonacciSession{
+		Contract: fib,
+		CallOpts: bind.CallOpts{
+			From: auth.From,
+		},
+		TransactOpts: *auth,
+	}
+
+	return fibonacciSession, nil
+}
+
+func TestFib(t *testing.T) {
+	session, err := RunValidators(t)
+	if err != nil {
+		t.Errorf("Validator setup error %v", err)
+	}
+
+	t.Run("TestFibResult", func(t *testing.T) {
+		fibsize := 15
+		fibnum := 11
+		fmt.Println("generating fib")
```

Comment:
Use t.Log or t.Logf instead of fmt.Print or log.Print variants when inside tests.